core/domain/collection_jobs_one_off.py

```diff
@@ -52,20 +58,26 @@ def map(item):
             collection.validate()
         except Exception as e:
             logging.error(
-                'Collection %s failed validation: %s' %
-                (item.id, e))
+                'Collection %s failed validation: %s' % (item.id, e))
+            yield (CollectionMigrationJob._ERROR_KEY,
+                   'Collection %s failed validation: %s' % (item.id, e))
```

Comment:
No need to log the exception since it's being logged with ``logging.error()`` and will be available via production logs. We can't always guarantee whether a particular Python value can be properly encoded into Unicode, and in the event this exception can't it will cause the mapreduce shard to fail (and a lot of errors to be outputted to the logs).